distribution/errors.go

```diff
@@ -84,7 +84,9 @@ func continueOnError(err error) bool {
 func retryOnError(err error) error {
 	switch v := err.(type) {
 	case errcode.Errors:
-		return retryOnError(v[0])
+		if len(v) != 0 {
+			return retryOnError(v[0])
+		}
```

Comment:
> this check won't be necessary after docker/distribution#1531 and distribution vendor is updated

Yep, was going to point this out. But it's defensive to have the check anyway, so I like this as-is.